internal/provider/resource_docker_image_funcs.go

```diff
@@ -319,6 +323,24 @@ func buildDockerImage(ctx context.Context, rawBuild map[string]interface{}, imag
 	buildOptions.Labels = labels
 	log.Printf("[DEBUG] Labels: %v\n", labels)
 
+	dockerClientVersion := client.ClientVersion()
+	log.Printf("[DEBUG] DockerClientVersion: %v, minBuildKitDockerVersion: %v\n", dockerClientVersion, minBuildkitDockerVersion)
+
+	if versions.GreaterThanOrEqualTo(dockerClientVersion, minBuildkitDockerVersion) {
+		// docker client supports BuildKit
+		log.Printf("[DEBUG] Enabling BuildKit")
+		s, _ := session.NewSession(ctx, "docker-provider", "")
```

Comment:
Return an error or disable BuildKit.

Reply:
I have taken the implementation from the hashicorp waypoint implementation and they also don't return an error there (https://github.com/hashicorp/waypoint/blob/main/builtin/docker/builder.go#L504)
I'm happy to implement, though :)

Reply:
Oh, I see.
session.NewSession always returns nil, so there is no problem for now.

https://github.com/moby/buildkit/blob/v0.10.3/session/session.go#L50-L88